src/main.rs

```diff
@@ -94,10 +94,48 @@ fn default_filename(filename: Option<String>) -> Result<PathBuf, clap::error::Er
     ))
 }
 
+fn assert_no_uncommitted_changes(path: &PathBuf) -> Result<(), clap::error::Error> {
+    // Extract the filename itself, as well as the directory from `path`.
+    assert!(path.is_file());
+    let filename_without_path = path.file_name().unwrap().to_str().unwrap();
+    let directory = path.parent().unwrap().to_str().unwrap();
+
+    let output = if cfg!(target_os = "windows") {
+        std::process::Command::new("cmd")
+            .args([
+                "/C",
+                format!("cd {}; git status --porcelain", directory).as_str(),
```

Comment:
This is stylistic, but since [String has `Deref<&str>`](https://doc.rust-lang.org/std/string/struct.String.html#impl-Deref-for-String), you don't need to call `.as_str()`: 

```suggestion
                &format!("cd {}; git status --porcelain", directory),
```